Gateway pod crashes when healthcheck is disabled.

**What happened**:
Gateway pod crashes when healthcheck is disabled.**What you expected to happen**:

[majopela@bluehat submariner]$ k logs submariner-gateway-jkknh
+ trap 'exit 1' SIGTERM SIGINT
+ export CHARON_PID_FILE=/var/run/charon.pid
+ CHARON_PID_FILE=/var/run/charon.pid
+ rm -f /var/run/charon.pid
+ SUBMARINER_VERBOSITY=2
+ '[' false == true ']'
+ DEBUG=-v=2
+ sysctl -w net.ipv4.conf.all.send_redirects=0
net.ipv4.conf.all.send_redirects = 0
+ exec submariner-gateway -v=2 -alsologtostderr
I0428 11:54:08.696796       1 main.go:90] Starting the submariner gateway engine
W0428 11:54:08.697072       1 client_config.go:543] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I0428 11:54:08.712407       1 main.go:118] Creating the cable engine
I0428 11:54:08.898505       1 local_endpoint.go:165] Interface "lo" has "127.0.0.1" address
I0428 11:54:08.898670       1 local_endpoint.go:165] Interface "br-ex" has "10.0.88.255" address
I0428 11:54:08.898838       1 local_endpoint.go:165] Interface "ovn-k8s-mp0" has "10.129.2.2" address
I0428 11:54:08.898847       1 local_endpoint.go:170] Found CNI Interface "ovn-k8s-mp0" that has IP "10.129.2.2" from ClusterCIDR "10.128.0.0/14"
I0428 11:54:08.898872       1 natdiscovery.go:114] NAT discovery server starting on port 0
I0428 11:54:08.898881       1 listener.go:33] NAT discovery protocol port not set for this gateway
I0428 11:54:08.898971       1 main.go:154] The CableEngine HealthChecker is disabled
I0428 11:54:08.898979       1 syncer.go:79] CableEngine syncer started
I0428 11:54:08.900115       1 main.go:296] Gateway leader election config values: main.leaderConfig{LeaseDuration:10, RenewDeadline:5, RetryPeriod:2}
I0428 11:54:08.900637       1 main.go:313] Using namespace "submariner-operator" for the leader election lock
I0428 11:54:08.900667       1 leaderelection.go:242] attempting to acquire leader lease  submariner-operator/submariner-gateway-lock...
I0428 11:54:08.914854       1 leaderelection.go:252] successfully acquired lease submariner-operator/submariner-gateway-lock
I0428 11:54:08.914991       1 event.go:281] Event(v1.ObjectReference{Kind:"ConfigMap", Namespace:"submariner-operator", Name:"submariner-gateway-lock", UID:"9befa61c-42cb-4d60-a5f9-71bcbc265f81", APIVersion:"v1", ResourceVersion:"110425", FieldPath:""}): type: 'Normal' reason: 'LeaderElection' ip-10-0-88-255-submariner-gateway became leader
I0428 11:54:08.915012       1 main.go:176] Creating the datastore syncer
I0428 11:54:08.915047       1 libreswan.go:95] Using NATT UDP port 4500
I0428 11:54:08.915268       1 libreswan.go:522] Starting Pluto
Initializing NSS database
nflog ipsec capture disabled
I0428 11:54:09.922083       1 cableengine.go:92] CableEngine controller started, driver: "libreswan"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x133b2ce]
goroutine 61 [running]:
main.main.func1.3(0xc000586fb0, 0x0, 0x0, 0xc0001ac000, 0xc0001883b0)
        command-line-arguments/main.go:210 +0x4e
created by main.main.func1
        command-line-arguments/main.go:207 +0x5b8

**How to reproduce it (as minimally and precisely as possible)**:
Disable g/w healthcheck
**Anything else we need to know?**:

**Environment**:
- Submariner version (use `subctl version`):
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

--- a/main.go
+++ b/main.go
@@ -207,8 +207,10 @@ func main() {
 		go func() {
 			defer wg.Done()
 
-			if err = cableHealthchecker.Start(stopCh); err != nil {
-				klog.Errorf("Error starting healthChecker: %v", err)
+			if cableHealthchecker != nil {
+				if err = cableHealthchecker.Start(stopCh); err != nil {
+					klog.Errorf("Error starting healthChecker: %v", err)
+				}
 			}
 		}()
 

